src/main/java/org/kuali/kfs/module/cam/web/struts/CapitalAssetInformationAction.java

```diff
@@ -103,6 +110,64 @@ protected void prepareRecordsForDisplay(CapitalAssetInformationForm capitalAsset
         capitalAssetForm.setCapitalAssetInformation(capitalAssetInformation);
     }
 
+    private boolean shouldBuildMissingCapitalAssetInformation(List<CapitalAssetInformation> capitalAssetInformations,
+            Collection<GeneralLedgerEntry> glEntries) {
+        boolean isCapitalAssetInformationAndGLEntriesDifferentSizes = !capitalAssetInformations.isEmpty() && capitalAssetInformations.size() != glEntries.size();
+        if (isGeneralErrorCorrectionDocument(glEntries)) {
+            return isCapitalAssetInformationAndGLEntriesDifferentSizes && isCapitalAssetInformationAndGLEntryTotalsDifferent(capitalAssetInformations, glEntries);
+        } else {
+            return isCapitalAssetInformationAndGLEntriesDifferentSizes;
+        }
+    }
+    
+    private boolean isGeneralErrorCorrectionDocument(Collection<GeneralLedgerEntry> glEntries) {
+        boolean isGeneralErrorCorrection = false;
+        if (CollectionUtils.isNotEmpty(glEntries)) {
+            GeneralLedgerEntry firstEntry = glEntries.iterator().next();
+            isGeneralErrorCorrection = StringUtils.equalsIgnoreCase(firstEntry.getFinancialDocumentTypeCode(), 
+                    KFSConstants.FinancialDocumentTypeCodes.GENERAL_ERROR_CORRECTION);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("isGeneralErrorCorrectionDocument: " + isGeneralErrorCorrection);
+        }
+        return isGeneralErrorCorrection;
+    }
+    
+    private boolean isCapitalAssetInformationAndGLEntryTotalsDifferent(List<CapitalAssetInformation> capitalAssetInformations,
+            Collection<GeneralLedgerEntry> glEntries) {
+        KualiDecimal entriesTotal = findTotalGLEntries(glEntries);
+        KualiDecimal informationTotal = findTotalAmountForAssetInformation(capitalAssetInformations);
+        if (LOG.isDebugEnabled()) {
+            int capAssetInfoSize = CollectionUtils.isNotEmpty(capitalAssetInformations) ? capitalAssetInformations.size() : 0;
+            int glEntreisSzie = CollectionUtils.isNotEmpty(glEntries) ? glEntries.size() : 0;
+            StringBuilder sb = new StringBuilder("isCapitalAssetInformationAndGLEntryTotalsDifferent, the number of capitalAssetInformations size: ");
+            sb.append(capAssetInfoSize).append(" glEntries size: ").append(glEntreisSzie);
+            sb.append(" informationTotal: ").append(informationTotal).append(" entriesTotal: ").append(entriesTotal);
+            LOG.debug(sb.toString());
+        }
+        return !entriesTotal.equals(informationTotal);
+    }
+    
+    private KualiDecimal findTotalAmountForAssetInformation(List<CapitalAssetInformation> informationList) {
+        KualiDecimal amount = KualiDecimal.ZERO;
+        if (CollectionUtils.isNotEmpty(informationList)) {
+            for (CapitalAssetInformation info : informationList) {
```

Comment:
I wonder if you could implement this using streams and reduce.

Reply:
good idea